### ISSUE TITLE
Rooms: Site

### ISSUE THREAD

**Issue by jlguillen** [SUNET/sri-front#273]
Can't be placed in a site while trying to create one

Can't be placed in a site while trying to edit one

**Comment by PaKZer0** [comment]
After some manual testing, it's upon creating a new room where it doesn't saves the designated site, but it stores it properly upon edition.

**Comment by mariahaider** [comment]
It still doesn't work. I chose the site, press save. But if I go back, click on the room again, i don't see the site i chose.

**Comment by PaKZer0** [comment]
Ok, so we've found out that there was a bug in the Sites Rooms' embed table. It was using the wrong relationship internally.
This is going to be fixed in the new release but there's still a chance that this doesn't solve the original problem of the task.

But what we can do is to test it in the new release and if the problem persist, get the data from the graphql query to figure out what's going on.

**Comment by mariahaider** [comment]
ok, sounds good.

**Comment by PaKZer0** [comment]
Ok, after another test round *the floor is not stored upon creation but it is on edition*. What we would do is to left this task unfinished (In progress) and we'll do the following:

- @mariahaider will test if an user is able to create/edit a room and add a site to it. We'll leave the floor field for the moment.
- @jlguillen will fix the storage of the floor field on creation, this should be fairly easy to do and test.

Thus this issue will be delivered on version 1.9.9

**Comment by mariahaider** [comment]
No, it is not working. I could add site from the noclook GUI but when I tried to delete it, i got 'server failure'.

![Screenshot 2021-01-28 at 12 28 11](https://user-images.githubusercontent.com/7047027/106132545-5ffd6d00-6164-11eb-80ed-7e4398cfe5cb.png)

But I noticed another thing, if I chose a site, then obviously the rack should be from the same site. Is it possible to see only those racks in a drop down menu like we see in noclook instead of searching them? I think the equipment should have the same mechanism. I asked @bergroth about it. He may reply here or I will get back to you if he thinks otherwise.

**Comment by PaKZer0** [comment]
> No, it is not working. I could add site from the noclook GUI but when I tried to delete it, i got 'server failure'.

This is caused for another reason, since the route for NOCLook changed from / to /api/ some routes are broken, I fix them as I see it in order to keep some kind of backwards compatibility, but there are some yet to be fixed (like this one).

>But I noticed another thing, if I chose a site, then obviously the rack should be from the same site. Is it possible to see only those racks in a drop down menu like we see in noclook instead of searching them? I think the equipment should have the same mechanism. I asked @bergroth about it. He may reply here or I will get back to you if he thinks otherwise.

You mean this kind of dropdowns?

![Screenshot_2021-01-28 NOCLook Edit](https://user-images.githubusercontent.com/2436503/106140268-d43d0e00-616e-11eb-88f3-4e4694a13e24.png)

AFAIK we ruled out having rooms being the parent of other rooms, we asked specifically this on the meeting we did with Magnus and the hierarchy we settled on was this:

<img width="980" alt="Captura de pantalla 2020-10-02 a las 10 49 39(1)" src="https://user-images.githubusercontent.com/2436503/106140438-12d2c880-616f-11eb-98b9-cf5582180aa0.png">

So what we have on SRI is:

- Sites that have (has relation) rooms and/or racks
- Rooms that have racks and are contained (parent relation) by a site (only one)
- Racks that are located (which are related with parent relation) by a room or a site (only one)

What we have on SRI is a reflect of this, thus for Rooms we have a Site table (that will hold only one parent Site) and a Rack table with multiple racks.

What I don't see the point on having these dropdowns to select a room as a parent of a room, or to show the racks contained by a Site, since what we want is to avoid breaking this relationship schema between these three location entities.

If we got something wrong about this, we would change how the location entities are related between themselves or how the ui lets the user build these relations but we need to agree beforehand on how this will work.

**Comment by PaKZer0** [comment]
> I think the equipment should have the same mechanism

In the other hand I agree that the location component for equipments now is far from perfect and it should be changed to match this hierarchy better and let the user locate any equipment on any of these three levels of locations.
But this development was postponed to do the rest of the feedback issues, there's no doubt that this would be an issue that we'll address in the next phase of the project if it suits the targets of it.

**Comment by mariahaider** [comment]
> > No, it is not working. I could add site from the noclook GUI but when I tried to delete it, i got 'server failure'.
> 
> This is caused for another reason, since the route for NOCLook changed from / to /api/ some routes are broken, I fix them as I see it in order to keep some kind of backwards compatibility, but there are some yet to be fixed (like this one).
> 
> > But I noticed another thing, if I chose a site, then obviously the rack should be from the same site. Is it possible to see only those racks in a drop down menu like we see in noclook instead of searching them? I think the equipment should have the same mechanism. I asked @bergroth about it. He may reply here or I will get back to you if he thinks otherwise.
> 
> You mean this kind of dropdowns?
> 
> ![Screenshot_2021-01-28 NOCLook Edit](https://user-images.githubusercontent.com/2436503/106140268-d43d0e00-616e-11eb-88f3-4e4694a13e24.png)
> 
> AFAIK we ruled out having rooms being the parent of other rooms, we asked specifically this on the meeting we did with Magnus and the hierarchy we settled on was this:
> 
> <img alt="Captura de pantalla 2020-10-02 a las 10 49 39(1)" width="980" src="https://user-images.githubusercontent.com/2436503/106140438-12d2c880-616f-11eb-98b9-cf5582180aa0.png">
> 
> So what we have on SRI is:
> 
> * Sites that have (has relation) rooms and/or racks
> * Rooms that have racks and are contained (parent relation) by a site (only one)
> * Racks that are located (which are related with parent relation) by a room or a site (only one)
> 
> What we have on SRI is a reflect of this, thus for Rooms we have a Site table (that will hold only one parent Site) and a Rack table with multiple racks.
> 
> What I don't see the point on having these dropdowns to select a room as a parent of a room, or to show the racks contained by a Site, since what we want is to avoid breaking this relationship schema between these three location entities.
> 
> If we got something wrong about this, we would change how the location entities are related between themselves or how the ui lets the user build these relations but we need to agree beforehand on how this will work.

ok understood!
